extensions/android-paging3/src/main/java/com/squareup/sqldelight/android/paging3/OffsetQueryPagingSource.kt

```diff
@@ -33,11 +33,13 @@ internal class OffsetQueryPagingSource<RowType : Any>(
   override suspend fun load(
     params: LoadParams<Long>
   ): LoadResult<Long, RowType> = withContext(dispatcher) {
+    val key = when {
+      params is LoadParams.Refresh -> (params.key ?: 0L) / params.loadSize * params.loadSize
```

Comment:
My Pager Config has a page size of 40. If I change the underlying data of item at key / index 3 and a `LoadParams.Refresh` is triggered. With the old behavior, data was reloaded however it was now using offset=3. This caused the recycler view to auto scroll for some reason that I don't understand. Also this wasn't happening with paging2.

After fiddling with it, I found out that when staying in the same offset range (offset=0) as earlier before, my item was instantly updated and there was no scrolling.